objects/builtin_json.go

```diff
@@ -0,0 +1,66 @@
+package objects
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+func builtinToJSON(args ...Object) (Object, error) {
+	if len(args) != 1 {
+		return nil, ErrWrongNumArguments
+	}
+
+	var toMarshal interface{}
+	var ok bool
+	switch o := args[0].(type) {
+	case *Array:
+		toMarshal, ok = ToSlice(o)
+		if !ok {
+			return nil, ErrInvalidTypeConversion
+		}
+	case *Map:
+		toMarshal, ok = ToMap(o)
+		if !ok {
+			return nil, ErrInvalidTypeConversion
+		}
+	default:
+		return nil, errors.New("to_json requires an array or map as argument")
```

Comment:
It does not have to be an array or a map. Primitive values are also valid JSON. https://play.golang.org/p/H7u25AhFsVZ